week2/index.js

```diff
@@ -0,0 +1,96 @@
+const fs = require('fs');
+const TODOS_FILENAME = process.env.TODOS_FILENAME || 'todo.json';
+let myTodoList = [];
+
+const readTodosFile = (filename) => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filename, "utf8", (err, data) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    console.log("Sorry, this file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    console.log("Sorry, but you don't have permission.");
+                } else {
+                    console.log(` unknown error occurred: ${err.message}`);
+                }
+                process.exit();
+            }
+            console.log(`Successfully read the file ${filename}`);
+            resolve(data);
+        });
+    });
+};
+const writeTodosFile = (filename, dataAsString) => {
+    return new Promise((resolve, reject) => {
+        fs.writeFile(filename, dataAsString, (err) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    console.log("Sorry, this file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    console.log("Sorry, but you don't have permission.");
+                } else {
+                    console.log(` unknown error occurred: ${err.message}`);
+                }
+                process.exit();
+            }
+            console.log(`${filename} file has been saved!`);
+            resolve();
+        });
+    });
+};
+
+const addTodoItem = (filename, todoList, itemText) => {
+    todoList.push({
+        message: itemText
+    });
+    writeTodosFile(TODOS_FILENAME, JSON.stringify(myTodoList)).then(() => console.log("Done adding todo."));
+};
+const removeTodoItem = (filename, todoList, itemIndex) => {
+    if (todoList.length > 0) {
+        todoList.splice(itemIndex - 1, 1);
+        writeTodosFile(TODOS_FILENAME, JSON.stringify(myTodoList)).then(() => console.log("Done Removing the task from your TODO list."));
```

Comment:
This line is too long and should be split up.

To avoid writing the same comment over again, please read Mo's homework feedback.

Reply:
I couldn't found Mo's..
Can you please explain what do you mean by saying split up..
do you mean . I have to put  .then() in new line?